Examples/Algorithms/TrackFinding/src/HoughTransformSeeder.cpp

```diff
@@ -0,0 +1,529 @@
+  }
+
+  std::vector<const SimSpacePoint*> spacePointPtrs;
+  spacePointPtrs.reserve(nSpacePoints);
+  for (const auto& isp : m_cfg.inputSpacePoints) {
+    for (auto& spacePoint : ctx.eventStore.get<SimSpacePointContainer>(isp)) {
+      // since the event store owns the space points, their pointers should be
+      // stable and we do noet need to create local copies.
+      spacePointPtrs.push_back(&spacePoint);
+    }
+  }
+
+  const auto& measurements =
+      ctx.eventStore.get<MeasurementContainer>(m_cfg.inputMeasurements);
+  const auto& sourceLinks =
+      ctx.eventStore.get<IndexSourceLinkContainer>(m_cfg.inputSourceLinks);
+
+  for (Acts::GeometryIdentifier geoId : m_cfg.geometrySelection) {
+    // select volume/layer depending on what is set in the geometry id
+    auto range = selectLowestNonZeroGeometryObject(sourceLinks, geoId);
+    // groupByModule only works with geometry containers, not with an
+    // arbitrary range. do the equivalent grouping manually
+    auto groupedByModule = makeGroupBy(range, detail::GeometryIdGetter());
+
+    for (auto [moduleGeoId, moduleSourceLinks] : groupedByModule) {
+      // find corresponding surface
+      const Acts::Surface* surface =
+          m_cfg.trackingGeometry->findSurface(moduleGeoId);
+      if (not surface) {
+        ACTS_ERROR("Could not find surface " << moduleGeoId);
+        return ProcessCode::ABORT;
+      }
+
+      for (auto& sourceLink : moduleSourceLinks) {
+        // extract a local position/covariance independent from the concrecte
+        // measurement content. since we do not know if and where the local
+        // parameters are contained in the measurement parameters vector, they
+        // are transformed to the bound space where we do know their location.
+        // if the local parameters are not measured, this results in a
+        // zero location, which is a reasonable default fall-back.
+        auto [localPos, localCov] = std::visit(
+            [](const auto& meas) {
+              auto expander = meas.expander();
+              Acts::BoundVector par = expander * meas.parameters();
+              Acts::BoundSymMatrix cov =
+                  expander * meas.covariance() * expander.transpose();
+              // extract local position
+              Acts::Vector2 lpar(par[Acts::eBoundLoc0], par[Acts::eBoundLoc1]);
+              // extract local position covariance.
+              Acts::SymMatrix2 lcov =
+                  cov.block<2, 2>(Acts::eBoundLoc0, Acts::eBoundLoc0);
+              return std::make_pair(lpar, lcov);
+            },
+            measurements[sourceLink.get().index()]);
+
+        // transform local position to global coordinates
+        Acts::Vector3 globalFakeMom(1, 1, 1);
+        Acts::Vector3 globalPos =
+            surface->localToGlobal(ctx.geoContext, localPos, globalFakeMom);
+        double r = sqrt(globalPos[Acts::ePos0] * globalPos[Acts::ePos0] +
```

Comment:
that can be conveniently replaced by `hypot(globalPos[Acts::ePos0], globalPos[Acts::ePos1])`